allennlp/commands/elmo.py

```diff
@@ -273,29 +275,28 @@ def embed_file(self,
         assert output_format in ["all", "top", "average"]
 
         # Tokenizes the sentences.
-        sentences = [line.strip() for line in input_file if line.strip()]
+        sentences = [line.strip() for line in input_file]
+        if "" in sentences:
+            raise ConfigurationError("Your input file contains empty lines. Please remove them.")
         split_sentences = [sentence.split() for sentence in sentences]
-        # Uses the sentence as the key.
-        embedded_sentences = zip(sentences, self.embed_sentences(split_sentences, batch_size))
+        # Uses the sentence index as the key.
+        embedded_sentences = enumerate(self.embed_sentences(split_sentences, batch_size))
 
         logger.info("Processing sentences.")
         with h5py.File(output_file_path, 'w') as fout:
             for key, embeddings in Tqdm.tqdm(embedded_sentences):
-                if key in fout.keys():
-                    logger.warning(f"Key already exists in {output_file_path}, skipping: {key}")
-                else:
-                    if output_format == "all":
-                        output = embeddings
-                    elif output_format == "top":
-                        output = embeddings[2]
-                    elif output_format == "average":
-                        output = numpy.average(embeddings, axis=0)
-
-                    fout.create_dataset(
-                            key,
-                            output.shape, dtype='float32',
-                            data=output
-                    )
+                if output_format == "all":
+                    output = embeddings
+                elif output_format == "top":
+                    output = embeddings[2]
```

Comment:
not sure if the whole command breaks if the weights file isn't 3 layers, but why not just use -1 here? (i know this isn't a change you introduced, just thought i'd bring it up)